prow/pod-utils/downwardapi/jobspec.go

```diff
@@ -175,23 +184,75 @@ func getRevisionFromRef(refs *prowapi.Refs) string {
 	return refs.BaseRef
 }
 
+func GetRevisionFromSpec(jobSpec *JobSpec) string {
+	return getRevisionFromRefs(jobSpec.Refs, jobSpec.ExtraRefs)
+}
+
 // GetRevisionFromSpec returns a main ref or sha from a spec object
-func GetRevisionFromSpec(spec *JobSpec) string {
-	if spec.Refs != nil {
-		return getRevisionFromRef(spec.Refs)
-	} else if len(spec.ExtraRefs) > 0 {
-		return getRevisionFromRef(&spec.ExtraRefs[0])
-	}
-	return ""
+func getRevisionFromRefs(refs *prowapi.Refs, extra []prowapi.Refs) string {
+	return getRevisionFromRef(mainRefs(refs, extra))
 }
 
-// MainRefs determines the main refs under test, if there are any
-func (s *JobSpec) MainRefs() *prowapi.Refs {
-	if s.Refs != nil {
-		return s.Refs
+func mainRefs(refs *prowapi.Refs, extra []prowapi.Refs) *prowapi.Refs {
+	if refs != nil {
+		return refs
 	}
-	if len(s.ExtraRefs) > 0 {
-		return &s.ExtraRefs[0]
+	if len(extra) > 0 {
+		return &extra[0]
 	}
 	return nil
 }
+
+func PjToStarted(pj *prowv1.ProwJob, cloneRecords []clone.Record) metadata.Started {
+	return refsToStarted(pj.Spec.Refs, pj.Spec.ExtraRefs, cloneRecords, pj.Status.StartTime.Unix())
+}
+
+func SpecToStarted(spec *JobSpec, cloneRecords []clone.Record) metadata.Started {
+	return refsToStarted(spec.Refs, spec.ExtraRefs, cloneRecords, time.Now().Unix())
+}
+
+// refsToStarted translate a jobspec into a started struct
```

Comment:
```suggestion
// refsToStarted translates refs into a Started struct.
```

Reply:
Oh I guess you're building this PR on top of the other one where you did the same thing here. nvm

Reply:
fixed